### PR TITLE
Cleaned up some un-free'd memory

### PR DESCRIPTION
**Pull request by neosilky**
Each commit contains the reason why each change was made. 

Found using Valgrind :)

### CODE REVIEW

File: sway/config.c [hunk 1]
```diff
@@ -267,6 +267,7 @@ static char *get_config_path(void) {
 		strcpy(config_home, home);
 		strcat(config_home, "/.config");
 		setenv("XDG_CONFIG_HOME", config_home, 1);
+		free(config_home);
 		sway_log(L_DEBUG, "Set XDG_CONFIG_HOME to %s", config_home);
```
Comment (mikkeloscar) [inline sway/config.c]: Move free to after `sway_log`.

File: sway/config.c [hunk 2]
```diff
@@ -532,6 +536,7 @@ bool read_config(FILE *file, struct sway_config *config) {
 		default:;
 		}
 		free(line);
+		free(res->input);
 		free(res);
```
Comment (mikkeloscar) [inline sway/config.c]: better to use `free_cmd_results` here.

File: sway/config.c (hunk 1 above)
Comment (neosilky) [inline sway/config.c]: Done (in latest commit)

File: sway/config.c (hunk 2 above)
Comment (neosilky) [inline sway/config.c]: Ah, didn't see that method. Done in the latest commit.

File: sway/config.c [hunk 3]
```diff
@@ -276,8 +277,9 @@ static char *get_config_path(void) {
 	int i;
 	for (i = 0; i < (int)(sizeof(config_paths) / sizeof(char *)); ++i) {
 		if (wordexp(config_paths[i], &p, 0) == 0) {
-			path = p.we_wordv[0];
+			path = strdup(p.we_wordv[0]);
 			if (file_exists(path)) {
+				wordfree(&p);
```
Comment (mikkeloscar) [inline sway/config.c]: `wordfree()` should be moved to before the if-statement, so it is called in all cases.

Comment (neosilky) [inline sway/config.c]: Also done :)

**Comment by mikkeloscar** [comment]
Thanks!

LGTM, but will let @SirCmpwn have a look.

**Comment by neosilky** [comment]
Great, let me know if you'd like me to squash the extra commits.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**